libcodechecker/server/api/report_server.py

```diff
@@ -237,7 +237,13 @@ def process_report_filter(session, report_filter):
             elif include:
                 file_ids = session.query(include_q.alias('include')).all()
 
-            OR.append(or_(File.id.in_([f_id[0] for f_id in file_ids])))
+            if file_ids:
+                OR.append(or_(File.id.in_([f_id[0] for f_id in file_ids])))
+            else:
+                # File id list can be empty for example when the user skips
+                # everything.
+                OR.append(False)
```

Comment:
Do we need to append `False` in this case, what if we do nothing if there are no file_ids?

Reply:
Yes, we have to. Let's see a dummy example: the user skips everything (`-*`) in the component. In this case the `file_ids` will be an empty list but we have to return with an empty result set because not the list of the component filter (`componentNames`) was empty.
We also have a test case for this: https://github.com/Ericsson/codechecker/blob/dc633732401c949b9d44ed166eb846791e4f92f2/tests/functional/component/test_component.py#L297